src/ashell/jerry-code.c

```diff
@@ -74,6 +120,100 @@ static void javascript_print_value(const jerry_value_t value)
     jerry_port_console("\n");
 }
 
+static int read_file(const char *file_name, char **file_buf)
+{
+    fs_file_t *fp = fs_open_alloc(file_name, "r");
+
+    if (fp == NULL)
+        return 0;
+
+    ssize_t size = fs_size(fp);
+    if (size == 0) {
+        comms_printf("[ERR] Empty file (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    *file_buf = (char *)zjs_malloc(size);
+    if (*file_buf == NULL) {
+        comms_printf("[ERR] Not enough memory for (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    ssize_t brw = fs_read(fp, *file_buf, size);
+
+    if (brw != size) {
+        comms_printf("[ERR] Failed loading code %s\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    return size;
+}
+
+static void find_requires(char *filebuf)
+{
+    char *ptr1 = filebuf;
+    char *ptr2 = NULL;
+    char *filestr = NULL;
+    char *ext_check = NULL;
+    while(ptr1 != NULL){
+
+        // Find next instance of "require"
+        ptr1 = strstr(ptr1, "require");
+
+        if (ptr1 != NULL) {
+            // Find the text between the two " " after require
+            ptr1 = strchr(ptr1, '"')+1;
+            ptr2 = strchr(ptr1, '"');
+            size_t len = ptr2-ptr1;
+
+            // Allocate the memory for the string
+            filestr = (char*)zjs_malloc(len+1);
+            strncpy(filestr, ptr1, len);
+            filestr[len] = '\0';
+
+            // Check that this is a *.js require and not a zephyr require
+            // Also check that the file hasn't already been loaded
+            ext_check = &filestr[len-3];
+
+            if (strcmp(ext_check, ".js") == 0 && !is_loaded(filestr))
+            {
+                push_to_list(&req_list, filestr);
+            }
+            zjs_free(filestr);
+            filestr = NULL;
+        }
+    }
+}
+
+static void load_require_modules(char *file_buffer)
+{
+    // Check if the file we've been asked to run requires any JS modules
+    find_requires(file_buffer);
+
+    // If it does, check if they have requires as well, and then eval the code
+    while (req_list) {
+        char *req_file_buffer = NULL;
+        uint file_size = read_file(req_list->file_name, &req_file_buffer);
+
+        if (file_size > 0)
+        {
+            find_requires(req_file_buffer);
```

Comment:
Hmm this is true.  I was really hoping to avoid having to open files multiple times, but it may be a necessary evil.  Thanks for the feedback.